static/app/utils/useSyncedLocalStorageState.tsx

```diff
@@ -0,0 +1,49 @@
+import {useCallback, useEffect} from 'react';
+
+import {useLocalStorageState} from 'sentry/utils/useLocalStorageState';
+
+type SyncedLocalStorageEvent = CustomEvent<{key: string; value: any}>;
```

Comment:
```suggestion
type SyncedLocalStorageEvent<S> = CustomEvent<{key: string; value: S}>;
```
You could then type guard between CustomEvent and SyncedLocalStorageEvent<S>